### PR TITLE
Create order API #28

### LINKED ISSUE #28: Make Order API

### PR DESCRIPTION
**Pull request by JeffKandel**
closes #28

### CODE REVIEW

File: server/orders.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    //
+    // If you want to only let admins list all the orders, then you'll
+    // have to add a role column to the orders table to support
+    // the concept of admin orders.
+    forbidden('listing orders is not allowed'),
+    (req, res, next) =>
+    Order.findAll()
+    .then(orders => res.json(orders))
+    .catch(next))
+  .post('/',
```
Comment (gabrielwr) [inline server/orders.js:21]: This whole block (line 17-20) is a little difficult on the eyes... perhaps some formatting/appropriate indenting is in order?

File: server/orders.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    (req, res, next) =>
+    Order.create(req.body)
+    .then(order => res.status(201).json(order))
+    .catch(next))
+  .get('/:id',
+    // TO DO: make sure that this order belongs to this user or user is Admin
+    mustBeLoggedIn,
```
Comment (gabrielwr) [inline server/orders.js:28]: Clarification -- I thought the requirements stated that "users" don't have to be loggedIn to place orders. Or am I thinking about the "cart"?

File: server/orders.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    (req, res, next) =>
+    Order.findById(req.params.id)
+    .then(order => res.json(order))
+    .catch(next))
+  .put('/:id',
+    // TO DO: make sure that this order belongs to this user or user is Admin
+    mustBeLoggedIn,
+    (req, res, next) =>
+    Order.findById(req.params.id)
```
Comment (gabrielwr) [inline server/orders.js:37]: Again, would nice to indent this so that it's a little easier to parse.

File: server/orders.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    (req, res, next) =>
+    Order.findAll()
+    .then(orders => res.json(orders))
+    .catch(next))
+  .post('/',
+    (req, res, next) =>
+    Order.create(req.body)
+    .then(order => res.status(201).json(order))
+    .catch(next))
+  .get('/:id',
```
Comment (gabrielwr) [inline server/orders.js:26]: We could use req.params here to dry any :id stuff out, but is certainly not necessary.

**Review by gabrielwr — COMMENTED** [review]
Looks good! See my comments for some formatting stuff.

File: server/api.js [hunk 5]
```diff
@@ -5,6 +5,7 @@ const api = module.exports = require('express').Router()
 api
   .get('/heartbeat', (req, res) => res.send({ok: true}))
   .use('/auth', require('./auth'))
+  .use('/orders', require('./orders'))
```
Comment (gabrielwr) [inline server/api.js:8]: isn't our model file called order.js?

**Review by gabrielwr — COMMENTED** [review]
2nd review changes, see all comments

File: server/api.js (hunk 5 above)
Comment (JeffKandel) [inline server/api.js]: No.

File: server/orders.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    mustBeLoggedIn,
+    (req, res, next) =>
+    Order.findById(req.params.id)
+    .then(order => order.destroy())
+    .then(wasDestroyedBool => {
+      if (wasDestroyedBool) {
+        res.sendStatus(204)
+      } else {
+        const err = Error('campus not destroyed')
+        err.status = 400
```
Comment (lada-tech) [inline server/orders.js]: Campus again?)

**Review by lada-tech — APPROVED** [review]
go ahead merge it

**Pull request merged**